### PR TITLE
#168781816 integrate request models

### PR DESCRIPTION
**Pull request by urbanishimwe**
#### What does this PR do?
- associate request model with trip model
- associate trip model with request model
- test for both models
#### Description of Task to be completed?
This PRs allows the application to create relationships among the aforementioned tables,
requests model will have all details associated to the user's request and it has a foreign key inside trips model which will help us to relate the trips with the request
#### How should this be manually tested?

*After cloning repo, installing required dependencies and database schema* then:
- add this variable in your `.env` if you didn't:
 1. DB_DEV= `link to your development database`
2. TEST_DB= `link to your test database`
- run this command `npm run db-migate:dev` in your command line to create all tables,
- you can use seeders to insert some dummy requests and use this routes to get requests of `alexismajyambere@gmail.com` (`npm run db-seeders:dev`):
   `localhost:3000/api/v1/requests/` (you need to have the token in the headers)
#### Any background context you want to provide?`
NAN
#### What are the relevant pivotal tracker stories?
[story](https://www.pivotaltracker.com/story/show/168781816)
#### Screenshots (if appropriate)
nan

#### Questions:
NAN

### CODE REVIEW

File: src/database/models/trip.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    Trips.belongsTo(models.Requests, {
+      foreignKey: {
+        allowNull: false,
+      },
+      onDelete: 'CASCADE',
+    });
+
+  };
+  return Trips;
+};
```
Comment (swaibat) [inline src/database/models/trip.js]: Good job @urbanishimwe  but add a new line at the end of this file

File: src/database/models/request.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    Requests.belongsTo(models.Users, {
+      foreignKey: {
+        allowNull: false,
+      },
+      onDelete: 'CASCADE',
+    });
+
+  };
+  return Requests;
+};
```
Comment (swaibat) [inline src/database/models/request.js]: Good job @urbanishimwe  but add a new line at the end of this file

File: src/tests/request.model.spec.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+/* eslint-disable max-len */
+import chai from 'chai';
+import db from '../database/models';
+
+const { expect } = chai;
+
+describe('request model tests', () => {
+  it('It should return all two way trip with length of two', async () => {
+    const trips = await db.Trips.count({ include: [{ model: db.Requests, where: { returnTime: { [db.Sequelize.Op.not]: null } } }] });
```
Comment (swaibat) [inline src/tests/request.model.spec.js]: Good job @urbanishimwe  but I don't think you should disable Eslint rule I think you should format your code according to the rule

**Review by swaibat — CHANGES_REQUESTED** [review]
@urbanishimwe  good job but code climate is failing

File: src/tests/request.model.spec.js (hunk 3 above)
Comment (urbanishimwe) [reply src/tests/request.model.spec.js]: If i do hound will fail

**Review by lemoissonM — CHANGES_REQUESTED** [review]
great job however you should fix the code climate issue

File: .codeclimate.yml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+exclude_patterns:
+- "src/tests/"
+- "src/database/"
+- "src/utils/"
```
Comment (swaibat) [inline .codeclimate.yml]: @urbanishimwe  awesome work, however, add a new line at the end of this file

**Review by swaibat — CHANGES_REQUESTED** [review]

**Review by swaibat — APPROVED** [review]

**Review by Crispy-rw — APPROVED** [review]

**Review by alexismajya — APPROVED** [review]

**Review by lemoissonM — APPROVED** [review]

File: src/controllers/requestController.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+/* eslint-disable max-len */
+import db from '../database/models';
+import helper from '../utils/helper';
+import sendResult from '../utils/sendResult';
+
+const Request = {
+  async getRequest(req, res) {
+    try {
+      const token = helper.getToken(req);
+      const { userId, verified } = await helper.verifyToken(token);
```
Comment (zellymiriam) [inline src/controllers/requestController.js]: Add a new line

File: src/controllers/requestController.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+      if (!(userId) || verified === undefined) {
+        return sendResult(res, 401, 'Invalid Token');
+      }
```
Comment (zellymiriam) [inline src/controllers/requestController.js]: Add a new line

File: src/controllers/requestController.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+      if (!verified) {
+        return sendResult(res, 400, 'The user is not verified');
+      }
```
Comment (zellymiriam) [inline src/controllers/requestController.js]: Add a new line

File: src/controllers/requestController.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+const Request = {
+  async getRequest(req, res) {
+    try {
+      const token = helper.getToken(req);
+      const { userId, verified } = await helper.verifyToken(token);
+      if (!(userId) || verified === undefined) {
+        return sendResult(res, 401, 'Invalid Token');
+      }
+      if (!verified) {
```
Comment (nakiwuge) [inline src/controllers/requestController.js]: Create a middleware for checking if a user is verified

**Review by nakiwuge — CHANGES_REQUESTED** [review]
Great job @urbanishimwe, I have left a few comments ob this PR. Kindly check them out

File: src/controllers/requestController.js [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+        return sendResult(res, 400, 'The user is not verified');
+      }
+      const trips = await db.Trips.findAll({ include: [{ model: db.Requests, where: { UserId: userId } }] });
```
Comment (nakiwuge) [inline src/controllers/requestController.js]: Add a new line here

File: src/controllers/requestController.js (hunk 8 above)
Comment (urbanishimwe) [reply src/controllers/requestController.js]: It is fixed

**Review by nakiwuge — APPROVED** [review]

**Pull request merged**